devito/data/allocators.py

```diff
@@ -128,8 +129,19 @@ class PosixAllocator(MemoryAllocator):
     @classmethod
     def initialize(cls):
         handle = find_library('c')
+
+        # Special case: on MacOS Big Sur any code that attempts to check
+        # for dynamic library presence by looking for a file at a path
+        # will fail. For this case, a static path is provided.
+        if handle is None and os.name == "posix" and sys.platform == "darwin":
+            handle = '/usr/lib/libc.dylib'
+
         if handle is not None:
-            cls.lib = ctypes.CDLL(handle)
+            try:
+                cls.lib = ctypes.CDLL(handle)
+            except OSError:
+                raise RuntimeError("Couldn't find `libc`'s `posix_memalign` to "
```

Comment:
no need to raise an error; just `cls.lib = None` and the error will be triggered, at a more suitable time (ie upon the first data allocation), by the next call to `_alloc_C_libcall`